pkg/services/datasources/service/datasource.go

```diff
@@ -534,21 +534,26 @@ func (s *Service) UpdateDataSource(ctx context.Context, cmd *datasources.UpdateD
 			}
 		}
 
-		// TODO: we will eventually remove this check for moving the resource to it's separate API
-		if s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders) && !cmd.OnlyUpdateLBACRulesFromAPI {
+		// TODO: Refactor to store lbac rules separate from a datasource
+		updatingLBACRules := cmd.OnlyUpdateLBACRulesFromAPI && s.features != nil && s.features.IsEnabled(ctx, featuremgmt.FlagTeamHttpHeaders) && cmd.JsonData != nil && cmd.JsonData.Get("teamHttpHeaders").Interface() != nil
+		if updatingLBACRules {
 			s.logger.Debug("Overriding LBAC rules with stored ones",
 				"reason", "update_lbac_rules_from_datasource_api",
 				"action", "use_updateLBACRules_API",
 				"datasource_id", dataSource.ID,
 				"datasource_uid", dataSource.UID)
-
+			cmd.JsonData.Set("teamHttpHeaders", cmd.JsonData.Get("teamHttpHeaders").Interface())
```

Comment:
Isn't this a no-op?

Reply:
yeas! this is correct. we do not actually need this step. same as you are saying further down in this PR - https://github.com/grafana/grafana/pull/94622#discussion_r1808271892.